SPTDataLoader/SPTDataLoaderRequestTaskHandler.m

```diff
@@ -239,11 +271,86 @@ - (void)completeIfInFlight
     }
 }
 
+#pragma mark KVO
+
+- (void)observeValueForKeyPath:(nullable NSString *)keyPath
+                      ofObject:(nullable id)object
+                        change:(nullable NSDictionary<NSString *, id> *)change
+                       context:(nullable  void *)context
+{
+    if (context == SPTDataLoaderRequestTaskHandlerContext) {
+        NSURLSessionTask *task = (NSURLSessionTask *)object;
+        if ([task isKindOfClass:[NSURLSessionTask class]]) {
+            id<SPTDataLoaderRequestResponseHandler> requestResponseHandler = self.requestResponseHandler;
+            if ([keyPath isEqualToString:kCountOfBytesReceived] || [keyPath isEqualToString:kCountOfBytesExpectedToReceive]) {
+                [requestResponseHandler updatedCountOfBytesReceived:self.task.countOfBytesReceived
+                                      countOfBytesExpectedToReceive:self.task.countOfBytesExpectedToReceive
+                                                         forRequest:self.request];
+            } else if ([keyPath isEqualToString:kCountOfBytesSent] || [keyPath isEqualToString:kCountOfBytesExpectedToSend]) {
+                [requestResponseHandler updatedCountOfBytesSent:self.task.countOfBytesSent
+                                     countOfBytesExpectedToSend:self.task.countOfBytesExpectedToSend
+                                                     forRequest:self.request];
+            }
+        }
+    }
```

Comment:
Should we call `[super observeValueForKeyPath:…]` in the `else` case?

Reply:
Calling super implementation is by the book but `SPTDataLoaderRequestTaskHandler` is the first class in the inheritance chain. So if anyone inherits from `SPTDataLoaderRequestTaskHandler` and makes its own default KVO implementation it will lead to the crash because calling `observeValueForKeyPath:ofObject:change:context:` on `NSObject` throws an `NSInternalInconsistency` exception.